### ISSUE TITLE
File works fine with appimage, but crashes with snap (unknown exception)

### ISSUE THREAD

**Issue by bambukouk** [realthunder/FreeCAD#404]
I am using:
```
OS: elementary OS 6.1 JÃ³lnir (Pantheon/pantheon)
Word size of FreeCAD: 64-bit
Version: 2022.430.26244 +4758 (Git) AppImage
Build type: Release
Branch: LinkDaily
Hash: b024b876480a533018fdc702145e9e38f8c54ba6
Python version: 3.9.12
Qt version: 5.12.9
Coin version: 4.0.1
OCC version: 7.5.3
Locale: English/United Kingdom (en_GB)
```
and:
```
OS: Ubuntu Core 20 (Pantheon/pantheon)
Word size of FreeCAD: 64-bit
Version: 2022.512.14555 (Git shallow)
Build type: Release
Branch: (HEAD detached at 2022.05.12-edge)
Hash: 8f837b53c108a2444535be2604fdf31170dcf7f0
Python version: 3.8.10
Qt version: 5.15.3
Coin version: 4.0.1
OCC version: 7.6.2
Locale: English/United Kingdom (en_GB)
```
File (renamed .zip to permit attaching here):
[02_loco_base(TEST).zip](https://github.com/realthunder/FreeCAD/files/8692856/02_loco_base.TEST.zip)

works perfectly fine with appimage version 2022.430.26244 +4758
but when I try to open it in snap version 2022.512.14555
it gives the following error message:
`12:26:00  4.5e-08 <App> Document.cpp(3210): Failed to restore _02_loco_base_TEST_#Sketch154: unknown exception
`
any attempts to open affected sketch (Sketch154) end up with FreeCAD crashing :-1: 

Help, please

**Comment by bambukouk** [comment]
OK, found what the problem was.
Shapebinder used in the affected sketch became corrupted after parent geometry was modified.
For some reason it was never highlighted in appimage version only becoming an issue when I used snap version.
Interestingly neither appimage nor snap showed any issues when I tried to validate geometry and sketch?

Easily solved, but...
1-why didn't it highlight as an issue in appimage version?
2-if I didn't have appimage I wouldn't be able to solve it (any attempts to do anything in snap, just crashed FreeCAD)

![Screenshot from 2022-05-16 09 01 27](https://user-images.githubusercontent.com/50301871/168549536-66aa9d1c-26fe-4ad2-b26b-5a09160597fc.png)
![Screenshot from 2022-05-16 09 09 55](https://user-images.githubusercontent.com/50301871/168549548-841f4b45-5f82-41ac-8a38-f2867a40212f.png)
